armv7a9-zynq7000/syspage.c

```diff
@@ -0,0 +1,501 @@
+	for (i = 0; i < syspage_common.mapsCnt; ++i)
+		low_memcpy((void *)((void *)syspage_common.syspage->maps + i * sizeof(syspage_map_t)), &syspage_common.maps[i].map, sizeof(syspage_map_t));
+
+	syspage_common.syspage->syspagesz += (syspage_common.mapsCnt * sizeof(syspage_map_t));
+	syspage_common.syspage->mapssz = syspage_common.mapsCnt;
+}
+
+
+/* TODO: align text and show map's attributes as string based on integer */
+void syspage_show(void)
+{
+	int i = 0;
+	plostd_printf(ATTR_LOADER, "\nSyspage addres: 0x%p\n", syspage_getAddress());
+	plostd_printf(ATTR_NONE, "--------------------------\n");
+
+	plostd_printf(ATTR_LOADER, "Kernel sections: \n");
+	plostd_printf(ATTR_USER, ".text: %p \tsize: %p\n", syspage_common.syspage->kernel.text, syspage_common.syspage->kernel.textsz);
+	plostd_printf(ATTR_USER, ".data: %p \tsize: %p\n", syspage_common.syspage->kernel.data, syspage_common.syspage->kernel.datasz);
+	plostd_printf(ATTR_USER, ".bss:  %p \tsize: %p\n", syspage_common.syspage->kernel.bss, syspage_common.syspage->kernel.bsssz);
+
+	plostd_printf(ATTR_LOADER, "\nPrograms number: %d\n", syspage_common.progsCnt);
+	if (syspage_common.progsCnt) {
+		plostd_printf(ATTR_LOADER, "NAME\t\tSTART\t\tEND\t\tIMAP\tDMAP\n");
+		for (i = 0; i < syspage_common.progsCnt; ++i)
+			plostd_printf(ATTR_USER, "%s\t\t%p\t%p\t%d\t%d\n", syspage_common.progs[i].cmdline, syspage_common.progs[i].start, syspage_common.progs[i].end, syspage_common.progs[i].imap, syspage_common.progs[i].dmap);
+	}
+
+	plostd_printf(ATTR_LOADER, "\n");
+	plostd_printf(ATTR_LOADER, "Mulimaps number: %d\n", syspage_common.mapsCnt);
+	if (syspage_common.mapsCnt) {
+		plostd_printf(ATTR_LOADER, "ID\tNAME\tSTART\t\tEND\tTOP\t\tFREESZ\tATTR\n");
+		for (i = 0; i < syspage_common.mapsCnt; ++i)
+			plostd_printf(ATTR_USER, "%d\t%s\t%p\t%p\t%p\t%d\t%d\n",  syspage_common.maps[i].map.id,  syspage_common.maps[i].map.name, syspage_common.maps[i].map.start,
+						  syspage_common.maps[i].map.end, syspage_common.maps[i].top, syspage_common.maps[i].map.end - syspage_common.maps[i].top, syspage_common.maps[i].map.attr);
+	}
+}
+
+
+
+/* Map's functions */
+
+int syspage_addmap(const char *name, void *start, void *end, u32 attr)
+{
+	int i;
+	u32 mapID, size;
+	mapID = syspage_common.mapsCnt;
+
+	/* Check whether map exists and overlaps with other maps */
+	for (i = 0; i < syspage_common.mapsCnt; ++i) {
+		if (((syspage_common.maps[i].map.start < (u32)end) && (syspage_common.maps[i].map.end > (u32)start)) ||
+			(plostd_strncmp(name, syspage_common.maps[i].map.name, plostd_strlen(syspage_common.maps[i].map.name)) == 0))
+			return -1;
+	}
+
+	syspage_common.maps[mapID].map.start = (u32)start;
+	syspage_common.maps[mapID].map.end = (u32)end;
+	syspage_common.maps[mapID].map.attr = attr;
+	syspage_common.maps[mapID].map.id = mapID;
+
+	size = plostd_strlen(name) < 7 ? plostd_strlen(name) : 7;
```

Comment:
```suggestion
	size = plostd_strlen(name);
	if (size > 7)
		size = 7;
```
One `plostd_strlen()` call instead two.